### PR TITLE
Improve hook args

### LINKED ISSUE #57: Enable hook to get access to test details and session id
Is this some how possible to write hooks to get access to test details like testid, associated desired capabilities and the session id on which the test ran so that it can be used as a trigger point for some related actions.

Please share if that is somehow possible now it self.

Thanks.

### PR DESCRIPTION
**Pull request by rsp9u**
* enable to pass the args if the length of args of the hook function is shortage
* pass SessionManager

Resolve #57

### CODE REVIEW

File: side_runner_py/utils.py [hunk 1]
```diff
@@ -22,13 +23,12 @@ def call_with_argname_bind(func, args_dict):
         if p.kind == Parameter.POSITIONAL_OR_KEYWORD
     ]
 
+    acceptable_args = {k: v for k, v in args_dict if k in param_names}
```
Comment (yaamai) [inline side_runner_py/utils.py]: maybe `args_dict` to `args_dict.items()`

**Review by yaamai — CHANGES_REQUESTED** [review]

File: side_runner_py/utils.py (hunk 1 above)
Comment (rsp9u) [reply side_runner_py/utils.py]: Fixed

**Review by yaamai — APPROVED** [review]

File: side_runner_py/hook.py [hunk 2]
```diff
@@ -45,38 +45,41 @@ def _run_hook(pre_or_post, kind, match_funcs, current_test_dict):
                 call_with_argname_bind(func, current_test_dict)
 
 
-def run_hook_per_project(pre_or_post, test_project):
+def run_hook_per_project(pre_or_post, session_manager, test_project):
```
Comment (qb-nimish) [inline side_runner_py/hook.py:48]: @rsp9u @yaamai May be session_manager can me kept as last argument and make it optional by setting it's default value to None?

Comment (qb-nimish) [reply side_runner_py/hook.py:48]: So that you don't have to modify all UTs

Comment (rsp9u) [reply side_runner_py/hook.py:48]: The instance of SessionManager lives while side-runner-py running. This argument is always passed, so I think it should be positional argument.

**Review by qb-nimish — APPROVED** [review]
LGTM

**Pull request merged**